relay-metrics/src/protocol.rs

```diff
@@ -349,20 +423,32 @@ fn parse_tags(string: &str) -> Option<BTreeMap<String, String>> {
 
 #[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
 pub struct Metric {
-    /// The name of the metric without its unit.
+    /// The metric resource identifier, in short MRI.
     ///
-    /// Metric names cannot be empty, must start with a letter and can consist of ASCII
-    /// alphanumerics, underscores, slashes, @s and periods.
-    pub name: String,
-    /// The unit of the metric value.
+    /// MRIs follow three core principles:
+
+    /// 1. **Robustness:** Metrics must be addressed via a stable identifier. During ingestion in
+    ///    Relay and Snuba, metrics are preaggregated and bucketed based on this identifier, so it
+    ///    cannot change over time without breaking bucketing.
+    /// 2. **Uniqueness:** The identifier for metrics must be unique across variations of units and
+    ///    metric types, within and across use cases, as well as between projects and
+    ///    organizations.
+    /// 3. **Abstraction:** The user-facing product changes its terminology over time, and splits
+    ///    concepts into smaller parts. The internal metric identifiers must abstract from that,
+    ///    and offer sufficient granularity to allow for such changes.
     ///
-    /// Units augment metric values by giving them a magnitude and semantics. There are certain
-    /// types of units that are subdivided in their precision, such as the [`DurationUnit`] for time
-    /// measurements.
+    /// MRIs have the format `<type>:<ns>/<name>@<unit>`, comprising the following components:
     ///
-    /// The unit can be omitted and defaults to [`MetricUnit::None`].
-    #[serde(default, skip_serializing_if = "MetricUnit::is_none")]
-    pub unit: MetricUnit,
+    /// * **Type:** counter (`c`), set (`s`), distribution (`d`), gauge (`g`), and evaluated (`e`) for derived numeric metrics. See [`MetricType`].
```

Comment:
We don't ingest these:
```suggestion
    /// * **Type:** counter (`c`), set (`s`), distribution (`d`), and gauge (`g`). See [`MetricType`].
```

Reply:
i updated the docstring to clarify that `e` is not observed in ingestion, but I kept it here